library/core/src/alloc/mod.rs

```diff
@@ -94,6 +94,18 @@ pub unsafe trait AllocRef {
     /// The returned block may have a larger size than specified by `layout.size()`, and may or may
     /// not have its contents initialized.
     ///
+    /// Note that you may not rely on this method actually getting called, even if there are calls
```

Comment:
I think this applies to `alloc_zeroed` as well, and the `GlobalAlloc::alloc*` methods. Maybe this should be a separate section somewhere.

Reply:
well, `alloc_zeroed` says it's the same as `alloc`, but with zeroing, so you need to read both docs anyway. Maybe this should just be on the `GlobalAlloc::alloc` method?